numpyro/distributions/distribution.py

```diff
@@ -58,6 +63,9 @@ def logpdf(self, x, *args, **kwargs):
 class jax_discrete(osp_stats.rv_discrete):
     args_check = True
 
+    def _support_mask(self, k):
+        return (k >= self.a) & (k <= self.b) & (np.floor(k) == k)
```

Comment:
I move this check outside in preparing for using a common support constraint instance as in pytorch